sklearn/preprocessing/_polynomial.py

```diff
@@ -1003,33 +1045,56 @@ def transform(self, X):
                     x = spl.t[spl.k] + (X[:, i] - spl.t[spl.k]) % (
                         spl.t[n] - spl.t[spl.k]
                     )
-                else:
+                else:  # self.extrapolation in ("continue", "error")
                     x = X[:, i]
 
                 if use_sparse:
+                    x = x.copy()  # copy to avoid inplace operation
+                    # we leave the nan values out of bsplining and make them into 0s
+                    # right away, since BSpline.design_matrix() would raise on the nans
+                    # when X is sparse:
+
+                    x[nan_indicator[:, i]] = np.nanmin(x)
                     XBS_sparse = BSpline.design_matrix(
                         x, spl.t, spl.k, **kwargs_extrapolate
                     )
+
                     if self.extrapolation == "periodic":
                         # See the construction of coef in fit. We need to add the last
                         # degree spline basis function to the first degree ones and
                         # then drop the last ones.
-                        # Note: See comment about SparseEfficiencyWarning below.
                         XBS_sparse = XBS_sparse.tolil()
                         XBS_sparse[:, :degree] += XBS_sparse[:, -degree:]
                         XBS_sparse = XBS_sparse[:, :-degree]
+
+                    # replace any indicated values with 0:
+                    extended_nan_indicator = np.repeat(
+                        nan_indicator[:, [i]], n_splines, axis=1
+                    )
+                    XBS_sparse[extended_nan_indicator] = 0
+
                 else:
                     XBS[:, (i * n_splines) : ((i + 1) * n_splines)] = spl(x)
+                    # replace any indicated values with 0:
+                    extended_nan_indicator = np.repeat(nan_indicator, n_splines, axis=1)
+                    XBS[extended_nan_indicator] = 0
+                    # adjust format of XBS to sparse, for scipy versions < 1.10.0:
+                    # TODO: Remove once scipy 1.10 is the minimum version:
+                    if self.sparse_output:
+                        XBS = sparse.csr_matrix(XBS)
```

Comment:
I am wondering about this Codecov failure, because this line is run when we use a scipy version below 1.10. In this case, `use_sparse` is `False` while `self.sparse_output `is `True`.

Reply:
I have just verified that when I make a local environment with scipy==1.8.0, and run the test, then it uses those two lines.